### ISSUE TITLE
usage is being printed even in the case of success

### ISSUE THREAD

**Issue by krancour** [Azure/azure-functions-core-tools#588]
Context:

OS: Mac OS 10.13.4

```
$ dotnet --version
2.1.302

$ func --version
2.0.1-beta.32
```

Usage is printed multiple times despite the command I entered appearing to be valid and exiting `0`.

Also worth noting are two apparent errors in the output containing "Could not find something to uninstall called ..."

```
$ func init --docker
Select a worker runtime: 
1. dotnet
2. node
3. java
Choose option: 1
dotnet
Usage: new [options]

Options:
  -h, --help          Displays help for this command.
  -l, --list          Lists templates containing the specified name. If no name is specified, lists all templates.
  -n, --name          The name for the output being created. If no name is specified, the name of the current directory is used.
  -o, --output        Location to place the generated output.
  -i, --install       Installs a source or a template pack.
  -u, --uninstall     Uninstalls a source or a template pack.
  --nuget-source      Specifies a NuGet source to use during install.
  --type              Filters templates based on available types. Predefined values are "project", "item" or "other".
  --force             Forces content to be generated even if it would change existing files.
  -lang, --language   Filters templates based on language and specifies the language of the template to create.


Templates                                         Short Name                          Language          Tags                                 
---------------------------------------------------------------------------------------------------------------------------------------------
QueueTrigger                                      Queue                               [C#]              Azure Function                       
HttpTrigger                                       Http                                [C#]              Azure Function                       
BlobTrigger                                       Blob                                [C#]              Azure Function                       
TimerTrigger                                      Timer                               [C#]              Azure Function                       
DurableFunctionsOrchestration                     DurableFunctionsOrchestration       [C#]              Azure Function                       
SendGrid                                          SendGrid                            [C#]              Azure Function                       
EventHubTrigger                                   EventHub                            [C#]              Azure Function                       
ServiceBusQueueTrigger                            SBQueue                             [C#]              Azure Function                       
ServiceBusTopicTrigger                            SBTopic                             [C#]              Azure Function                       
EventGridTrigger                                  EventGrid                           [C#]              Azure Function                       
IotHubTrigger                                     IotHub                              [C#]              Azure Function                       
CosmosDBTrigger                                   CosmosDB                            [C#]              Azure Function                       
Azure Functions                                   azureFunctionsProjectTemplates      [C#], F#          AzureFunctions/ClassLib              
Console Application                               console                             [C#], F#, VB      Common/Console                       
Class library                                     classlib                            [C#], F#, VB      Common/Library                       
Unit Test Project                                 mstest                              [C#], F#, VB      Test/MSTest                          
xUnit Test Project                                xunit                               [C#], F#, VB      Test/xUnit                           
Razor Page                                        page                                [C#]              Web/ASP.NET                          
MVC ViewImports                                   viewimports                         [C#]              Web/ASP.NET                          
MVC ViewStart                                     viewstart                           [C#]              Web/ASP.NET                          
ASP.NET Core Empty                                web                                 [C#], F#          Web/Empty                            
ASP.NET Core Web App (Model-View-Controller)      mvc                                 [C#], F#          Web/MVC                              
ASP.NET Core Web App                              razor                               [C#]              Web/MVC/Razor Pages                  
ASP.NET Core with Angular                         angular                             [C#]              Web/MVC/SPA                          
ASP.NET Core with React.js                        react                               [C#]              Web/MVC/SPA                          
ASP.NET Core with React.js and Redux              reactredux                          [C#]              Web/MVC/SPA                          
Razor Class Library                               razorclasslib                       [C#]              Web/Razor/Library/Razor Class Library
ASP.NET Core Web API                              webapi                              [C#], F#          Web/WebAPI                           
global.json file                                  globaljson                                            Config                               
NuGet Config                                      nugetconfig                                           Config                               
Web Config                                        webconfig                                             Config                               
Solution File                                     sln                                                   Solution                             

Examples:
    dotnet new mvc --auth Individual
    dotnet new webapi 
    dotnet new --help
Usage: new [options]

Options:
  -h, --help          Displays help for this command.
  -l, --list          Lists templates containing the specified name. If no name is specified, lists all templates.
  -n, --name          The name for the output being created. If no name is specified, the name of the current directory is used.
  -o, --output        Location to place the generated output.
  -i, --install       Installs a source or a template pack.
  -u, --uninstall     Uninstalls a source or a template pack.
  --nuget-source      Specifies a NuGet source to use during install.
  --type              Filters templates based on available types. Predefined values are "project", "item" or "other".
  --force             Forces content to be generated even if it would change existing files.
  -lang, --language   Filters templates based on language and specifies the language of the template to create.


Templates                                         Short Name                          Language          Tags                                 
---------------------------------------------------------------------------------------------------------------------------------------------
QueueTrigger                                      Queue                               [C#]              Azure Function                       
HttpTrigger                                       Http                                [C#]              Azure Function                       
BlobTrigger                                       Blob                                [C#]              Azure Function                       
TimerTrigger                                      Timer                               [C#]              Azure Function                       
DurableFunctionsOrchestration                     DurableFunctionsOrchestration       [C#]              Azure Function                       
SendGrid                                          SendGrid                            [C#]              Azure Function                       
EventHubTrigger                                   EventHub                            [C#]              Azure Function                       
ServiceBusQueueTrigger                            SBQueue                             [C#]              Azure Function                       
ServiceBusTopicTrigger                            SBTopic                             [C#]              Azure Function                       
EventGridTrigger                                  EventGrid                           [C#]              Azure Function                       
IotHubTrigger                                     IotHub                              [C#]              Azure Function                       
CosmosDBTrigger                                   CosmosDB                            [C#]              Azure Function                       
Azure Functions                                   azureFunctionsProjectTemplates      [C#], F#          AzureFunctions/ClassLib              
Console Application                               console                             [C#], F#, VB      Common/Console                       
Class library                                     classlib                            [C#], F#, VB      Common/Library                       
Unit Test Project                                 mstest                              [C#], F#, VB      Test/MSTest                          
xUnit Test Project                                xunit                               [C#], F#, VB      Test/xUnit                           
Razor Page                                        page                                [C#]              Web/ASP.NET                          
MVC ViewImports                                   viewimports                         [C#]              Web/ASP.NET                          
MVC ViewStart                                     viewstart                           [C#]              Web/ASP.NET                          
ASP.NET Core Empty                                web                                 [C#], F#          Web/Empty                            
ASP.NET Core Web App (Model-View-Controller)      mvc                                 [C#], F#          Web/MVC                              
ASP.NET Core Web App                              razor                               [C#]              Web/MVC/Razor Pages                  
ASP.NET Core with Angular                         angular                             [C#]              Web/MVC/SPA                          
ASP.NET Core with React.js                        react                               [C#]              Web/MVC/SPA                          
ASP.NET Core with React.js and Redux              reactredux                          [C#]              Web/MVC/SPA                          
Razor Class Library                               razorclasslib                       [C#]              Web/Razor/Library/Razor Class Library
ASP.NET Core Web API                              webapi                              [C#], F#          Web/WebAPI                           
global.json file                                  globaljson                                            Config                               
NuGet Config                                      nugetconfig                                           Config                               
Web Config                                        webconfig                                             Config                               
Solution File                                     sln                                                   Solution                             

Examples:
    dotnet new mvc --auth Individual
    dotnet new azureFunctionsProjectTemplates --StorageConnectionStringValue --AzureFunctionsVersion
    dotnet new --help
The template "Azure Functions" was created successfully.


Could not find something to uninstall called '/usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.32/templates/itemTemplates.nupkg'.
Usage: new [options]

Options:
  -h, --help          Displays help for this command.
  -l, --list          Lists templates containing the specified name. If no name is specified, lists all templates.
  -n, --name          The name for the output being created. If no name is specified, the name of the current directory is used.
  -o, --output        Location to place the generated output.
  -i, --install       Installs a source or a template pack.
  -u, --uninstall     Uninstalls a source or a template pack.
  --nuget-source      Specifies a NuGet source to use during install.
  --type              Filters templates based on available types. Predefined values are "project", "item" or "other".
  --force             Forces content to be generated even if it would change existing files.
  -lang, --language   Filters templates based on language and specifies the language of the template to create.


Templates                                         Short Name                          Language          Tags                                 
---------------------------------------------------------------------------------------------------------------------------------------------
QueueTrigger                                      Queue                               [C#]              Azure Function                       
HttpTrigger                                       Http                                [C#]              Azure Function                       
BlobTrigger                                       Blob                                [C#]              Azure Function                       
TimerTrigger                                      Timer                               [C#]              Azure Function                       
DurableFunctionsOrchestration                     DurableFunctionsOrchestration       [C#]              Azure Function                       
SendGrid                                          SendGrid                            [C#]              Azure Function                       
EventHubTrigger                                   EventHub                            [C#]              Azure Function                       
ServiceBusQueueTrigger                            SBQueue                             [C#]              Azure Function                       
ServiceBusTopicTrigger                            SBTopic                             [C#]              Azure Function                       
EventGridTrigger                                  EventGrid                           [C#]              Azure Function                       
IotHubTrigger                                     IotHub                              [C#]              Azure Function                       
CosmosDBTrigger                                   CosmosDB                            [C#]              Azure Function                       
Azure Functions                                   azureFunctionsProjectTemplates      [C#], F#          AzureFunctions/ClassLib              
Console Application                               console                             [C#], F#, VB      Common/Console                       
Class library                                     classlib                            [C#], F#, VB      Common/Library                       
Unit Test Project                                 mstest                              [C#], F#, VB      Test/MSTest                          
xUnit Test Project                                xunit                               [C#], F#, VB      Test/xUnit                           
Razor Page                                        page                                [C#]              Web/ASP.NET                          
MVC ViewImports                                   viewimports                         [C#]              Web/ASP.NET                          
MVC ViewStart                                     viewstart                           [C#]              Web/ASP.NET                          
ASP.NET Core Empty                                web                                 [C#], F#          Web/Empty                            
ASP.NET Core Web App (Model-View-Controller)      mvc                                 [C#], F#          Web/MVC                              
ASP.NET Core Web App                              razor                               [C#]              Web/MVC/Razor Pages                  
ASP.NET Core with Angular                         angular                             [C#]              Web/MVC/SPA                          
ASP.NET Core with React.js                        react                               [C#]              Web/MVC/SPA                          
ASP.NET Core with React.js and Redux              reactredux                          [C#]              Web/MVC/SPA                          
Razor Class Library                               razorclasslib                       [C#]              Web/Razor/Library/Razor Class Library
ASP.NET Core Web API                              webapi                              [C#], F#          Web/WebAPI                           
global.json file                                  globaljson                                            Config                               
NuGet Config                                      nugetconfig                                           Config                               
Web Config                                        webconfig                                             Config                               
Solution File                                     sln                                                   Solution                             

Examples:
    dotnet new mvc --auth Individual
    dotnet new IotHub --namespace --Connection --Path messages/events
    dotnet new --help
Could not find something to uninstall called '/usr/local/Cellar/azure-functions-core-tools/2.0.1-beta.32/templates/projectTemplates.nupkg'.
Usage: new [options]

Options:
  -h, --help          Displays help for this command.
  -l, --list          Lists templates containing the specified name. If no name is specified, lists all templates.
  -n, --name          The name for the output being created. If no name is specified, the name of the current directory is used.
  -o, --output        Location to place the generated output.
  -i, --install       Installs a source or a template pack.
  -u, --uninstall     Uninstalls a source or a template pack.
  --nuget-source      Specifies a NuGet source to use during install.
  --type              Filters templates based on available types. Predefined values are "project", "item" or "other".
  --force             Forces content to be generated even if it would change existing files.
  -lang, --language   Filters templates based on language and specifies the language of the template to create.


Templates                                         Short Name                          Language          Tags                                 
---------------------------------------------------------------------------------------------------------------------------------------------
QueueTrigger                                      Queue                               [C#]              Azure Function                       
HttpTrigger                                       Http                                [C#]              Azure Function                       
BlobTrigger                                       Blob                                [C#]              Azure Function                       
TimerTrigger                                      Timer                               [C#]              Azure Function                       
DurableFunctionsOrchestration                     DurableFunctionsOrchestration       [C#]              Azure Function                       
SendGrid                                          SendGrid                            [C#]              Azure Function                       
EventHubTrigger                                   EventHub                            [C#]              Azure Function                       
ServiceBusQueueTrigger                            SBQueue                             [C#]              Azure Function                       
ServiceBusTopicTrigger                            SBTopic                             [C#]              Azure Function                       
EventGridTrigger                                  EventGrid                           [C#]              Azure Function                       
IotHubTrigger                                     IotHub                              [C#]              Azure Function                       
CosmosDBTrigger                                   CosmosDB                            [C#]              Azure Function                       
Azure Functions                                   azureFunctionsProjectTemplates      [C#], F#          AzureFunctions/ClassLib              
Console Application                               console                             [C#], F#, VB      Common/Console                       
Class library                                     classlib                            [C#], F#, VB      Common/Library                       
Unit Test Project                                 mstest                              [C#], F#, VB      Test/MSTest                          
xUnit Test Project                                xunit                               [C#], F#, VB      Test/xUnit                           
Razor Page                                        page                                [C#]              Web/ASP.NET                          
MVC ViewImports                                   viewimports                         [C#]              Web/ASP.NET                          
MVC ViewStart                                     viewstart                           [C#]              Web/ASP.NET                          
ASP.NET Core Empty                                web                                 [C#], F#          Web/Empty                            
ASP.NET Core Web App (Model-View-Controller)      mvc                                 [C#], F#          Web/MVC                              
ASP.NET Core Web App                              razor                               [C#]              Web/MVC/Razor Pages                  
ASP.NET Core with Angular                         angular                             [C#]              Web/MVC/SPA                          
ASP.NET Core with React.js                        react                               [C#]              Web/MVC/SPA                          
ASP.NET Core with React.js and Redux              reactredux                          [C#]              Web/MVC/SPA                          
Razor Class Library                               razorclasslib                       [C#]              Web/Razor/Library/Razor Class Library
ASP.NET Core Web API                              webapi                              [C#], F#          Web/WebAPI                           
global.json file                                  globaljson                                            Config                               
NuGet Config                                      nugetconfig                                           Config                               
Web Config                                        webconfig                                             Config                               
Solution File                                     sln                                                   Solution                             

Examples:
    dotnet new mvc --auth Individual
    dotnet new webconfig
    dotnet new --help
Writing /Users/kent/kentsfirstserverlessapp/.vscode/extensions.json
fatal: Not a git repository (or any of the parent directories): .git
Initialized empty Git repository in /Users/kent/kentsfirstserverlessapp/.git/
```

```
$ echo $?
0
```

**Comment by krancour** [comment]
Update: This issue seems only to occur with .Net as the runtime.

**Comment by ahmelsayed** [comment]
This was fixed in the same commit as #589
